### PR TITLE
BUG: Fix AssertionError for numpy bool index in _handle_index_argument

### PR DESCRIPTION
**Pull request by dhruv1955**
### Summary
Removes an internal `assert` in `_handle_index_argument` that caused 
`AssertionError` instead of the intended `ValueError` when a numpy 
bool was passed as the `index` argument.

### What changed
- Removed `assert index is True` from `_handle_index_argument` in 
  `astropy/table/_dataframes.py`
- Kept the existing user-facing `ValueError` for missing single-column 
  primary key
- Added a regression test in `TestDataFrameConversion` 
  (`astropy/table/tests/test_df.py`)

### Why
`np.bool_(True) in (None, True)` evaluates to `True`, but 
`np.bool_(True) is True` is `False`, so the assert fired and raised 
`AssertionError` instead of the expected `ValueError`. This made 
behavior inconsistent and confusing for users passing numpy booleans.

### Testing
Added regression test:
- `test_to_df_index_numpy_bool_true_without_primary_key`
- Verifies `ValueError` is raised with correct message when 
  `index=np.bool_(True)` is passed without a single-column primary key

### CODE REVIEW

**Comment by dhruv1955** [comment]
Hi! This is my third Astropy PR (previous: #19351 merged, #19357 open). The codecov drop is expected — the removed `assert` line no longer exists to be covered. Happy to address any review feedback. Thanks!

File: astropy/table/tests/test_df.py [hunk 1]
```diff
@@ -328,6 +328,24 @@ def __init__(self, *args, **kwargs):
         with pytest.raises(ValueError, match="is not in the table columns"):
             self._to_dataframe(t, backend, use_legacy_pandas_api, index="not a column")
 
+    def test_to_df_index_numpy_bool_true_without_primary_key(
+        self, backend, use_legacy_pandas_api
+    ):
+        if backend != "pandas":
+            return
```
Comment (pllim) [inline astropy/table/tests/test_df.py]: Shouldn't we skip with reason instead of return here?

File: astropy/table/tests/test_df.py [hunk 2]
```diff
@@ -328,6 +328,24 @@ def __init__(self, *args, **kwargs):
...
+        t = table.QTable([[1, 2], [3, 4]], names=["a", "b"])
+
+        with pytest.raises(
+            ValueError, match="index=True requires a single-column primary key"
+        ):
+            self._to_dataframe(
+                t,
+                backend,
+                use_legacy_pandas_api,
+                index=np.bool_(True),
```
Comment (pllim) [inline astropy/table/tests/test_df.py:346]: Why `np.bool_(True)` and not just `True`?

File: docs/changes/table/19358.bugfix.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+Fixed `AssertionError` when passing `numpy.bool_(True)` as the `index` argument to `Table.to_pandas()` or `Table.to_dataframe()`. The correct `ValueError` is now raised instead.
```
Comment (pllim) [inline docs/changes/table/19358.bugfix.rst]: Please use double backticks to prevent intersphinx of kicking in and breaking doc build.

File: astropy/table/tests/test_df.py (hunk 1 above)
Comment (dhruv1955) [reply astropy/table/tests/test_df.py]: Fixed — replaced `return` with `pytest.skip("requires pandas")`.

File: astropy/table/tests/test_df.py (hunk 2 above)
Comment (dhruv1955) [reply astropy/table/tests/test_df.py:346]: `np.bool_(True)` is necessary because `True is True` is `True`, so the old `assert index is True` would not fire for plain `True`. The failure only appears with `np.bool_(True)` where `np.bool_(True) is True` is `False` even though it is truthy. This test specifically guards that edge case.

File: docs/changes/table/19358.bugfix.rst (hunk 3 above)
Comment (dhruv1955) [reply docs/changes/table/19358.bugfix.rst]: Fixed — updated to double backticks throughout.

**Comment by pllim** [comment]
Thanks. Let's wait for subpackage maintainer to review, in case there was a good reason why they added `assert` into production code.

File: docs/changes/table/19358.bugfix.rst [hunk 4]
```diff
@@ -0,0 +1 @@
+Fixed ``AssertionError`` when passing ``numpy.bool_(True)`` as the ``index`` argument to ``Table.to_pandas()`` or ``Table.to_dataframe()``. The correct ``ValueError`` is now raised instead.
```
Comment (neutrinoceros) [inline docs/changes/table/19358.bugfix.rst]: I think (though I might misremember) that we usally enforce a 80-chars limit on these. Not crucial though.
```suggestion
Fixed ``AssertionError`` when passing ``numpy.bool_(True)`` as the
``index`` argument to ``Table.to_pandas()`` or ``Table.to_dataframe()``.
The correct ``ValueError`` is now raised instead.
```

**Comment by dhruv1955** [comment]
Thanks @neutrinoceros  Wrapped the changelog entry at 80 chars and squashed to a single commit.

**Comment by pllim** [comment]
RTD failed. You cannot auto merge. FYI.

**Comment by dhruv1955** [comment]
> RTD failed. You cannot auto merge. FYI.

@pllim the RTD failure appears unrelated to this PR - the only files changed are `astropy/table/_dataframes.py`, `astropy/table/tests/test_df.py`, and `docs/changes/table/19358.bugfix.rst`. No documentation pages were modified. Would it be possible to retry the RTD build?

File: docs/changes/table/19358.bugfix.rst
Comment (pllim) [inline docs/changes/table/19358.bugfix.rst:1]: This is why RTD is failing

```
generating indices... <unknown>:169: WARNING: py:obj reference target not found: numpy.bool_(True) [ref.obj]
<unknown>:169: WARNING: py:obj reference target not found: index [ref.obj]
<unknown>:169: WARNING: py:obj reference target not found: Table.to_pandas() [ref.obj]
<unknown>:169: WARNING: py:obj reference target not found: Table.to_dataframe() [ref.obj]
```

Comment (pllim) [reply docs/changes/table/19358.bugfix.rst:1]: Hint: Use double backticks.

Comment (pllim) [reply docs/changes/table/19358.bugfix.rst:1]: In fact, we discourage single backticks in change log to prevent broken links in the future.

**Comment by dhruv1955** [comment]
pre-commit.ci autofix

**Comment by dhruv1955** [comment]
@neutrinoceros @pllim all checks are now passing including RTD. Could you re-enable auto-merge? Thanks!

**Comment by pllim** [comment]
I'll defer to @neutrinoceros (please **squash and merge**)
